src/storage/src/render/upsert/types.rs

```diff
@@ -132,10 +130,26 @@ pub struct PutValue<V> {
 ///
 /// This struct is not part of the `UpsertStateBackend` public API, but implementing that API without
 /// using it is considered hard-mode.
+///
+///
+/// Note also that this type is designed to support _partial updates_. All values are
+/// associated with an _ordered_ `O` that can be used to determine if a value existing in the
+/// `UpsertStateBackend` occurred before or after a value being considered for insertion.
+/// `O` is typically required to be `: Default`, with the default value sorting below all others.
```

Comment:
The `O` in this case is actually an `Option<Partitioned>`, so the `default() == None` case serializes much smaller!

Reply:
That doesn't feel right because as a user of the upsert machinery I'm instantiating it with a timestamp and the fact that an `Option` is used is an implementation detail for performance reasons, so I would expect this to be `Option<O>` and have `O: Timestamp`